core/adapters/presto.ts

```diff
@@ -0,0 +1,92 @@
+import { IAdapter } from "df/core/adapters";
+import { Adapter } from "df/core/adapters/base";
+import { Task, Tasks } from "df/core/tasks";
+import { dataform } from "df/protos/ts";
+
+export class PrestoAdapter extends Adapter implements IAdapter {
+  constructor(private readonly project: dataform.IProjectConfig, dataformCoreVersion: string) {
+    super(dataformCoreVersion);
+  }
+
+  public resolveTarget(target: dataform.ITarget) {
+    // Database here is equivalent to Presto's catalog.
+    return `"${target.database || this.project.defaultDatabase}"."${target.schema ||
+      this.project.defaultSchema}"."${target.name}"`;
+  }
+
+  public publishTasks(
+    table: dataform.ITable,
+    runConfig: dataform.IRunConfig,
+    tableMetadata?: dataform.ITableMetadata
+  ): Tasks {
+    const tasks = Tasks.create();
+
+    this.preOps(table, runConfig, tableMetadata).forEach(statement => tasks.add(statement));
+
+    const baseTableType = this.baseTableType(table.type);
+    if (tableMetadata && tableMetadata.type !== baseTableType) {
+      tasks.add(
+        Task.statement(this.dropIfExists(table.target, this.oppositeTableType(baseTableType)))
+      );
+    }
+
+    if (table.type === "incremental") {
+      throw new Error("Incremental table types are not currently supported for Presto.");
+    } else {
+      tasks.add(Task.statement(this.createOrReplace(table)));
+    }
+
+    this.postOps(table, runConfig, tableMetadata).forEach(statement => tasks.add(statement));
+
+    if (runConfig.useSingleQueryPerAction) {
```

Comment:
side note: we should probably get rid of this option now that it's been defaulted on for a long while!